vpnglean/ios.cmake

```diff
@@ -0,0 +1,118 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# When building for iOS we want to include the Glean iOS SDK on the build as well.
+# Instead of using it out of the box, we want it to be linked to the vpnglean_ffi target,
+# so we generate the SDK oursevels.
+
+enable_language(Swift)
+
+set(GLEAN_VENDORED_PATH ${CMAKE_SOURCE_DIR}/3rdparty/glean)
+
+add_library(iosglean SHARED)
+
+target_include_directories(iosglean PUBLIC ${CMAKE_SOURCE_DIR})
+target_include_directories(iosglean PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(iosglean PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/glean)
+
+set_target_properties(iosglean PROPERTIES
+    FRAMEWORK TRUE
+    FRAMEWORK_VERSION C
+    OUTPUT_NAME "Glean"
+    MACOSX_FRAMEWORK_INFO_PLIST "${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Info.plist"
+    MACOSX_FRAMEWORK_IDENTIFIER "${BUILD_IOS_APP_IDENTIFIER}.Glean"
+    XCODE_ATTRIBUTE_SWIFT_VERSION "5.0"
+    XCODE_ATTRIBUTE_CLANG_ENABLE_MODULES "YES"
+    XCODE_ATTRIBUTE_APPLICATION_EXTENSION_API_ONLY "YES"
+    XCODE_ATTRIBUTE_MACH_O_TYPE "staticlib"
+    XCODE_ATTRIBUTE_SWIFT_OBJC_BRIDGING_HEADER "${GLEAN_VENDORED_PATH}/glean-core/ios/Glean/Glean.h"
```

Comment:
On the prototype (https://github.com/mozilla-mobile/mozilla-vpn-client/pull/4596) I did not use a bridging header. However here I could not get the application to compile and find the stuff defined in the header files without doing so... Not sure what is the difference, clearly something is different on the config, but the build system change is hiding that for me. cc @mcleinman